packages/@react-aria/progress/src/useProgressBar.ts

```diff
@@ -50,9 +50,9 @@ export function useProgressBar(props: AriaProgressBarProps): ProgressBarAria {
 
   value = clamp(value, minValue, maxValue);
   let percentage = (value - minValue) / (maxValue - minValue);
-  let formatter = useNumberFormatter(formatOptions);
 
   if (!isIndeterminate && !valueLabel) {
+    let formatter = useNumberFormatter(formatOptions);
```

Comment:
This unfortunately breaks the rules of hooks. However, within useNumberFormatter we do cache them, so over the lifetime of an app, the expense should become less.